instrumentation-api-appender/src/main/java/io/opentelemetry/instrumentation/api/appender/LogEmitter.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.appender;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A {@link LogEmitter} is the entry point into a log pipeline.
+ *
+ * <p>Obtain a log builder via {@link #logBuilder()}, add properties using the setters, and emit it
+ * via {@link LogBuilder#emit()}.
+ */
+@ThreadSafe
```

Comment:
We have a deep precedent of adding `@ThreadSafe` to interfaces, but it continues to be weird to me because it's just a hint and it's always possible to implement a very thread unsafe implementation of these interfaces. 🤷🏻